### DISCUSSION TITLE
Show country in table column

### DISCUSSION THREAD

**Discussion by medfrk** [parfaitementweb/filament-country-field#1] [Ideas]
I use filament-country-field to show a select box to choose a list of country but and the selected value will be saved as country code in db. any way to show the full country name in table column.
thank a loot

**Comment by AlexisSerneels** [comment]
Hey @medfrk The TableColumn is something I want to implement as well. Will do.
Or you are always welcome to send a PR meanwhile ;)

**Reply by medfrk** [reply]
Thank you for your response! I appreciate your willingness to implement TableColumn. I'll look forward to the update. If I come across any potential solutions or improvements, I'll be sure to share them. Thanks again for your help! 😊

**Comment by marcellodum** [comment]
Not very elegant but this is my workaround:
 ```
Tables\Columns\TextColumn::make('country')
    ->label(__('Nazione'))
    ->formatStateUsing(function($state):string {
        $country = new Country('tmp_country');
        $countryList = $country->getList();
        return $countryList[$state] ?? $state;
    })
    ->searchable(),
 ```

**Comment by AlexisSerneels** [comment]
The latest version included the TableColumn ;)
Enjoy.

https://github.com/parfaitementweb/filament-country-field/releases/tag/2.1.0
